be/src/olap/generic_iterators.cpp

```diff
@@ -148,22 +150,23 @@ class MergeIteratorContext {
     RowBlockV2 _block;
 
     bool _valid = false;
-    size_t _index_in_block = 0;
+    size_t _index_in_block = -1;
 };
 
 Status MergeIteratorContext::init(const StorageReadOptions& opts) {
     RETURN_IF_ERROR(_iter->init(opts));
     RETURN_IF_ERROR(_load_next_block());
+    if (valid()) {
+        RETURN_IF_ERROR(advance());
```

Comment:
why this place to get another block, it may be consume more memory in init()

Reply:
here should seek to the first filtered row in the current block, not to get another block.